cob_script_server/src/simple_script_server/simple_script_server.py

```diff
@@ -711,6 +709,13 @@ def set_light(self,parameter_name,blocking=False):
 
 		rospy.loginfo("Set light to <<%s>>",parameter_name)
 		
+		topic_parameter_name = self.ns_global_prefix + "/light/topic_name"
+		if not rospy.has_param(topic_parameter_name):
+			rospy.logerr("parameter %s does not exist on ROS Parameter Server, aborting...",topic_parameter_name)
+			return 2
+		self.light_topic_name = rospy.get_param(topic_parameter_name)
+		self.pub_light = rospy.Publisher(self.light_topic_name, ColorRGBA, queue_size=1)
```

Comment:
this is critical as mostly the first message on a new topic is lost. this is why the publisher previously was created in __init__.

better solution: use service instead of topic. get topic name from self.ns_global_prefix + "/" + component_name + "service_ns"